tests/test_unittest/test_gtc/test_passes/test_oir_optimizations/test_horizontal_execution_merging.py

```diff
@@ -172,3 +172,28 @@ def test_on_the_fly_merging_repeated():
     transformed = OnTheFlyMerging().visit(testee)
     hexecs = transformed.vertical_loops[0].sections[0].horizontal_executions
     assert len(hexecs) == 2
+
+
+def test_on_the_fly_merging_nested():
```

Comment:
Is this testing the LocalScalar declaration problem? How?

Reply:
Yes, it is a nested merging. The first merge step merges the first horizontal iteration into 2-4, then the second merges what was the second into the remaining two. It ends up with `tmp0` defined in two `HorizontalExecution`s.